src/main/java/io/micronaut/build/MicronautQualityChecksParticipantPlugin.java

```diff
@@ -0,0 +1,27 @@
+package io.micronaut.build;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.jvm.JvmTestSuite;
+import org.gradle.testing.base.TestingExtension;
+import org.gradle.testing.jacoco.plugins.JacocoPlugin;
+
+@SuppressWarnings("UnstableApiUsage")
+public class MicronautQualityChecksParticipantPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(final Project project) {
+        configureJacoco(project);
+    }
+
+    private void configureJacoco(final Project project) {
+        project.getPluginManager().withPlugin("java", p -> {
+            project.getPluginManager().apply(JacocoPlugin.class);
+
+            final TestingExtension testing = project.getExtensions().findByType(TestingExtension.class);
+            if (testing != null) {
+                testing.getSuites().withType(JvmTestSuite.class, JvmTestSuite::useJUnitJupiter);
```

Comment:
This is Gradle 7.4 only, right? A few questions:

- does this mean that jacoco reports won't work if we don't use JUnit Jupiter?
- the base plugin already configures test suites to use JUnit Jupiter, so this might be redundant
- since not all test suites are using the new 7.4 API, if we need this, either we should restrict to the test suite we actually care about (say, `tasks.named("test") { ... }`), or it should be a global flag `tasks.withType(Test).configureEach { ... }`